examples/moses-pln-synergy/scm/README.md

```diff
@@ -76,16 +76,204 @@ And search for the following
    )
 ...
 
-scheme@(guile-user)> (cog-bind pln-rule-eliminate-neutral-element-hack)
-scheme@(guile-user)> (cog-bind pln-rule-eliminate-dangling-junctor-hack)
-scheme@(guile-user)> (cog-bind pln-rule-equivalence-hack)
-$6 = (SetLink
+scheme@(guile-user)> (cog-bind implication-lambda-distribution-rule)
+...
    (ImplicationLink (stv 1 0.99999982)
-      (PredicateNode "take-treatment-1" (stv 0.1 0.80000001))
-      (PredicateNode "take-compound-A" (stv 0.2 0.80000001))
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (AndLink
+            (EvaluationLink
+               (PredicateNode "take")
+               (ListLink
+                  (VariableNode "$X")
+                  (ConceptNode "treatment-1")
+               )
+            )
+            (EvaluationLink (stv 1 0.99999982)
+               (PredicateNode "contain")
+               (ListLink
+                  (ConceptNode "treatment-1")
+                  (ConceptNode "compound-A")
+               )
+            )
+         )
+      )
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
```

Comment:
What I meant was this: In predicate logic, if one has that
```
ForAll x (P(x) -> Q(x))
```
then one can deduce that it holds for any single choice of `x` that is, that 
```
P(x) -> Q(x)
```
In the above, x is free.   You can make it bound, by writing
```
lambda y (P(y)->Q(y))
```
Lets go back to the first expression, lets call it `A`  that is `A` is the expression `ForAll x (P(x) -> Q(x))`  Now, in classical predicate logic, `A` would be either true or false. In Bayesian logic, yes, assigning a 'probability' of  `Sum_x (P(x) and Q(x)) / Sum_y P(y)` to `A` is the correct answer.  (Please observe my careful use of the bound variable `y` in the above.) Note that this assignment of this truth-value to `A`  predates the arrival of either you or I or Ben on this planetary surface. I believe that this formula can be found in even old texts; its "obvious".  

Now suppose that `L` stands for `lambda y (P(y)->Q(y))` -- the question then is: what is the truth value of `L`?  You seem to be stating that it has exactly the same truth value as `A` does !??  That is not at all clear to me. 

Consider now the expression `M` which is `P(x)->Q(x)`  What is the truth value for `M`?  It seems plainly clear to me that both `L` and `M` have the same truth value, but you seem to disagree about that too.  By the way, observe that `M = L(x)`  which is hopefully obvious, and needs no explanation, right? So of course both `L` and `M` should have the same truth value ... how could they not?  Yet you reject `M` and embrace `L`, and I don't understand why.  They are just two sides of the same coin.

Finally, consider the opencog expression 
```
ImplicationLink <TV>
   P
   Q
```
You seem to claim that this is identical to `L`  which is fine by me, but then you insist that it has a truth-value that is identical to `A`.   And then you claim that `M` cannot even have a truth value.   This is just one of many, mulltiple confusions.